webapp/components/_new/generic/BaseButton/BaseButton.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <button
+    :class="buttonClass"
+    :disabled="loading"
+    :type="type"
+    @click.capture="event => $emit('click', event)"
+  >
+    <base-icon v-if="icon" :name="icon" />
+    <loading-spinner v-if="loading" />
+    <slot />
+  </button>
+</template>
+
+<script>
+import LoadingSpinner from '~/components/_new/generic/LoadingSpinner/LoadingSpinner'
+
+export default {
+  components: {
+    LoadingSpinner,
+  },
+  props: {
+    circle: {
+      type: Boolean,
+      default: false,
+    },
+    danger: {
+      type: Boolean,
+      default: false,
+    },
+    filled: {
+      type: Boolean,
+      default: false,
+    },
+    ghost: {
+      type: Boolean,
+      default: false,
+    },
+    icon: {
+      type: String,
+    },
+    loading: {
+      type: Boolean,
+      default: false,
+    },
+    size: {
+      type: String,
+      default: 'regular',
+      validator(value) {
+        return value.match(/(small|regular)/)
+      },
+    },
+    type: {
+      type: String,
+      default: 'button',
+      validator(value) {
+        return value.match(/(button|submit)/)
+      },
+    },
+  },
+  computed: {
+    buttonClass() {
+      let buttonClass = 'base-button'
+
+      if (this.$slots.default == null) buttonClass += ' --icon-only'
```

Comment:
To also return `true` when it's `undefined`. But I just double-checked and I think slots are `undefined` when they're empty so it could be `=== undefined` instead. :+1: